File: java/javafind/src/test/java/javafind/FileTypesTest.java
package javafind;

import org.junit.jupiter.api.Test;

import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.*;

public class FileTypesTest {
    private final FileTypes fileTypes;

    public FileTypesTest() {
        fileTypes = new FileTypes();
    }

    @Test
    public final void archiveFileTest() {
        var path = Paths.get("test.zip");
        assertEquals("zip", FileUtil.getExtension(path));
        assertTrue(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.ARCHIVE, fileTypes.getFileType(path));
    }

    @Test
    public final void audioFileTest() {
        var path = Paths.get("music.mp3");
        assertEquals("mp3", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertTrue(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.AUDIO, fileTypes.getFileType(path));
    }

    @Test
    public final void binaryFileTest() {
        var path = Paths.get("test.exe");
        assertEquals("exe", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertTrue(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.BINARY, fileTypes.getFileType(path));
    }

    @Test
    public final void codeFileTest() {
        var path = Paths.get("Test.java");
        assertEquals("java", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertTrue(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertTrue(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.CODE, fileTypes.getFileType(path));
    }

    @Test
    public final void fontFileTest() {
        var path = Paths.get("font.ttf");
        assertEquals("ttf", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertTrue(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.FONT, fileTypes.getFileType(path));
    }

    @Test
    public final void imageFileTest() {
        var path = Paths.get("Test.png");
        assertEquals("png", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertTrue(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.IMAGE, fileTypes.getFileType(path));
    }

    @Test
    public final void textFileTest() {
        var path = Paths.get("test.txt");
        assertEquals("txt", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertTrue(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.TEXT, fileTypes.getFileType(path));
    }

    @Test
    public final void videoFileTest() {
        var path = Paths.get("movie.mp4");
        assertEquals("mp4", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertTrue(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertEquals(FileType.VIDEO, fileTypes.getFileType(path));
    }

    @Test
    public final void xmlFileTest() {
        var path = Paths.get("markup.xml");
        assertEquals("xml", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertTrue(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertTrue(fileTypes.isXmlFile(path));
        assertEquals(FileType.XML, fileTypes.getFileType(path));
    }

    @Test
    public final void unknownFileTest() {
        var path = Paths.get("unknown.ZZZ");
        assertEquals("zzz", FileUtil.getExtension(path));
        assertFalse(fileTypes.isArchiveFile(path));
        assertFalse(fileTypes.isAudioFile(path));
        assertFalse(fileTypes.isBinaryFile(path));
        assertFalse(fileTypes.isCodeFile(path));
        assertFalse(fileTypes.isFontFile(path));
        assertFalse(fileTypes.isImageFile(path));
        assertFalse(fileTypes.isTextFile(path));
        assertFalse(fileTypes.isVideoFile(path));
        assertFalse(fileTypes.isXmlFile(path));
        assertTrue(fileTypes.isUnknownFile(path));
        assertEquals(FileType.UNKNOWN, fileTypes.getFileType(path));
    }
}

File: kotlin/ktfind/src/test/kotlin/ktfind/FileTypesTest.kt
package ktfind

import java.io.File
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

/**
 * @author cary on 7/30/16.
 */
class FileTypesTest {
    private val fileTypes = FileTypes()

    @Test
    fun archiveFileTest() {
        val file = File("test.zip")
        assertEquals("zip", file.extension)
        assertTrue(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.ARCHIVE, fileTypes.getFileType(file))
    }

    @Test
    fun audioFileTest() {
        val file = File("music.mp3")
        assertEquals("mp3", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertTrue(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.AUDIO, fileTypes.getFileType(file))
    }

    @Test
    fun binaryFileTest() {
        val file = File("test.exe")
        assertEquals("exe", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertTrue(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.BINARY, fileTypes.getFileType(file))
    }

    @Test
    fun fontFileTest() {
        val file = File("font.ttf")
        assertEquals("ttf", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.FONT, fileTypes.getFileType(file))
    }

    @Test
    fun imageFileTest() {
        val file = File("Test.png")
        assertEquals("png", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertTrue(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.IMAGE, fileTypes.getFileType(file))
    }

    @Test
    fun javaFileTest() {
        val file = File("Test.java")
        assertEquals("java", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertTrue(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.CODE, fileTypes.getFileType(file))
    }

    @Test
    fun textFileTest() {
        val file = File("test.txt")
        assertEquals("txt", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.TEXT, fileTypes.getFileType(file))
    }

    @Test
    fun textFileTestUppercase() {
        val file = File("TEXT.TXT")
        assertEquals("TXT", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.TEXT, fileTypes.getFileType(file))
    }

    @Test
    fun videoFileTest() {
        val file = File("movie.mp4")
        assertEquals("mp4", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertTrue(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.VIDEO, fileTypes.getFileType(file))
    }

    @Test
    fun unknownFileTest() {
        val file = File("unknown.ZZZ")
        assertEquals("ZZZ", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isAudioFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isFontFile(file))
        assertFalse(fileTypes.isImageFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isVideoFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertTrue(fileTypes.isUnknownFile(file))
        assertEquals(FileType.UNKNOWN, fileTypes.getFileType(file))
    }
}
